libs/wingsdk/src/target-sim/bucket.inflight.ts

```diff
@@ -57,10 +59,7 @@ export class Bucket implements IBucketClient, ISimulatorResourceInstance {
     return this.context.withTrace({
       message: `List (prefix=${prefix ?? "null"}).`,
       activity: async () => {
-        const fileNames = await fs.promises.readdir(this.fileDir);
-        return prefix
-          ? fileNames.filter((fileName) => fileName.startsWith(prefix))
-          : fileNames;
+        return fs.promises.readdir(`${this.fileDir}/${prefix ?? ""}`);
```

Comment:
Does `list` read all files out recursively (even in sub-directories)? I think we need to add a unit test case like this to match up with how AWS's ListObjects API works (and other clouds too):

```ts
await bucket.put("foo");
await bucket.put("foo/bar");
await bucket.put("foo/bar/baz");
const resp = await bucket.list(); // no prefix
expect(resp).toEqual([
  "foo",
  "foo/bar",
  "foo/bar/baz",
]);
```

Reply:
It appears that the behavior of the list may be influenced by whether it represents `s3api list-objects` or `s3 list`. 
In my opinion, considering that the AWS s3 console likely utilizes `s3 list`, I would anticipate a similar behavior in the wing console. Nonetheless, I am a newcomer to this platform, and you possess a better understanding of the system, so please feel free to correct me if I am wrong.

Reply:
For now I think we will start with modeling after `s3api list-objects` as it's the lower-level of the two APIs. (`s3 list` can be implemented by calling `list-objects` and using the `delimiter` option, and then formatting the output a bit). There's some more discussion about the API here: https://github.com/winglang/wing/issues/1673

In any case it's probably fine to handle this issue separately / it doesn't need to be part of this PR, so I'll approve 😄